src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -442,7 +442,7 @@ public override ModelPropertyCollection Properties
                 if (_properties == null)
                 {
                     var properties = _provider.GetMetadataForProperties(ModelType);
-                    properties = properties.OrderBy(p => p.Order);
+                    properties = properties.OrderBy(p => p.Order).ThenBy(p => p.PropertyName);
```

Comment:
OrdinalIgnoreCase is good for exact comparisons but not for sorting strings with unusual characters. This is also faster because there are no extra rules to apply before comparing.